packages/upload-client/src/index.js

```diff
@@ -129,15 +130,17 @@ async function uploadBlockStream(conf, blocks, options = {}) {
           Store.add(conf, bytes, options),
           (async () => {
             const hasher = PieceHasher.create()
-            const digestBytes = new Uint8Array(36)
+            const digestBytes = new Uint8Array(PIECE_MULTIHASH_SIZE)
             hasher.write(bytes)
             hasher.digestInto(digestBytes, 0, true)
-            const digest = Digest.create(PieceHasher.code, digestBytes)
-            return Link.create(raw.code, digest)
-          })()
+            const digest = Digest.decode(digestBytes)
```

Comment:
@alanshaw `digestBytes` already contained multihash prefix and `Digest.create` was adding another prefix.  I have changed to `Digest.decode` instead which will attempt to interprent given bytes as multihash instead.